src/semidiscretization/semidiscretization_hyperbolic_parabolic.jl

```diff
@@ -0,0 +1,181 @@
+
+# TODO: Taal refactor, Mesh<:AbstractMesh{NDIMS}, Equations<:AbstractEquations{NDIMS} ?
+"""
+    SemidiscretizationHyperbolicParabolic
+
+A struct containing everything needed to describe a spatial semidiscretization
+of a hyperbolic-parabolic conservation law.
+"""
+struct SemidiscretizationHyperbolicParabolic{SemiHyperbolic, SemiParabolic, Cache} <: AbstractSemidiscretization
+  semi_hyperbolic::SemiHyperbolic
+  semi_parabolic::SemiParabolic
+  cache::Cache
+  performance_counter::PerformanceCounter
+
+  function SemidiscretizationHyperbolicParabolic{SemiHyperbolic, SemiParabolic, Cache}(
+      semi_hyperbolic::SemiHyperbolic, semi_parabolic::SemiParabolic,
+      cache::Cache) where {SemiHyperbolic, SemiParabolic, Cache}
+
+    performance_counter = PerformanceCounter()
+
+    new(semi_hyperbolic, semi_parabolic, cache, performance_counter)
+  end
+end
+
+"""
+    SemidiscretizationHyperbolicParabolic(semi_hyperbolic, semi_parabolic)
+
+Construct a semidiscretization of a hyperbolic-parabolic PDE by combining the purely hyperbolic and
+purely parabolic semidiscretizations.
+"""
+function SemidiscretizationHyperbolicParabolic(semi_hyperbolic, semi_parabolic)
+  cache = (; du_ode_parabolic=Vector{real(semi_parabolic)}())
+
+  SemidiscretizationHyperbolicParabolic{typeof(semi_hyperbolic), typeof(semi_parabolic), typeof(cache)}(
+    semi_hyperbolic, semi_parabolic, cache)
+end
+
+
+function Base.show(io::IO, semi::SemidiscretizationHyperbolicParabolic)
+  print(io, "SemidiscretizationHyperbolicParabolic(")
+  print(io,       semi.semi_hyperbolic)
+  print(io, ", ", semi.semi_parabolic)
+  print(io, ", cache(")
+  for (idx,key) in enumerate(keys(semi.cache))
+    idx > 1 && print(io, " ")
+    print(io, key)
+  end
+  print(io, "))")
+end
+
+function Base.show(io::IO, ::MIME"text/plain", semi::SemidiscretizationHyperbolicParabolic)
+  if get(io, :compact, false)
+    show(io, semi)
+  else
+    summary_header(io, "SemidiscretizationHyperbolicParabolic")
+    summary_line(io, "hyperbolic system", semi.semi_hyperbolic)
+    summary_line(io, "parabolic system", semi.semi_parabolic)
+    summary_footer(io)
+  end
+end
+
+
+@inline Base.ndims(semi::SemidiscretizationHyperbolicParabolic) = ndims(semi.semi_hyperbolic.mesh)
+
+@inline nvariables(semi::SemidiscretizationHyperbolicParabolic) = nvariables(semi.semi_hyperbolic.equations)
+
+@inline Base.real(semi::SemidiscretizationHyperbolicParabolic) = real(semi.semi_hyperbolic.solver)
+
+
+@inline function mesh_equations_solver_cache(semi::SemidiscretizationHyperbolicParabolic)
+  @unpack mesh, equations, solver, cache = semi.semi_hyperbolic
+  return mesh, equations, solver, cache
```

Comment:
```suggestion
  return mesh_equations_solver_cache(semi.semi_hyperbolic)
```
This allows more flexibility

Reply:
Thanks!